### ISSUE TITLE
check-types should accept capitalized primitive types

### ISSUE THREAD

**Issue by ZacBrownBand** [gajus/eslint-plugin-jsdoc#25]
Eslint and this plugin conflict.
Eslint suggest that `{string}` should be `{String}`
This plugin suggests that `{String}` should be `{string}`

Jscs setting would be `"checkTypes": "capitalizedNativeCase",`

**Comment by gajus** [comment]
This has been discussed a number of times, e.g. https://github.com/gajus/eslint-plugin-jsdoc/issues/22

ESLint is wrong to recommend `String`. 99% of the time you mean `string`, e.g.

```js
// Thats 'string'
const foo = 'FOO';

// Thats 'String'
const bar = new String('bar');
```

**Comment by ZacBrownBand** [comment]
Right, so `{String}` is a valid type, regardless of if its not the common case.
Could there be an option to allow for these types? Like a `nativeTypes` vs [`strictNativeTypes`](https://github.com/gajus/eslint-plugin-jsdoc/blob/master/src/rules/checkTypes.js#L32)

**Comment by gajus** [comment]
It is such a rare edge case to use these types that I would like to avoid adding support for it. I have not seen a single valid use of a native type constructor in the past 5 years.

**Comment by Redsandro** [comment]
Ah I get it. `{String}` and `{string}` are technically both valid, but they are not the same.

```js
new String('lard') === 'lard' // false

new String('lard') // String {0: "l", 1: "a", 2: "r", 3: "d", length: 4}
'lard' // "lard"
```

​So it's not about consistency, rather about probability of use.

type name | `typeof` | check-types | testcase
--|--|--|--
**Object** | object | **Object** | `({}) instanceof Object` -> `true`
**Array** | object | **Array** | `([]) instanceof Array` -> `true`
**Date** | object | **Date** | `(new Date()) instanceof Date` -> `true`
**RegExp** | object | **RegExp** | `(new RegExp(/.+/)) instanceof RegExp` -> `true`
Boolean | **boolean** | **boolean** | `(true) instanceof Boolean` -> **`false`**
Number | **number** | **number** | `(41) instanceof Number` -> **`false`**
String | **string** | **string** | `("test") instanceof String` -> **`false`**

Confusing. But makes sense.

**Comment by gajus** [comment]
@Redsandro Please contribute a variation of this to the documentation. This is the most often asked question.

**Comment by Redsandro** [comment]
@gajus https://github.com/gajus/eslint-plugin-jsdoc/pull/30
